core/src/test/java/feign/client/DefaultClientTest.java

```diff
@@ -151,6 +152,39 @@ public void retriesFailedHandshake() throws IOException, InterruptedException {
     assertEquals(2, server.getRequestCount());
   }
 
+  @Test
+  public void safeRebuffering() throws IOException, InterruptedException {
+    server.enqueue(new MockResponse().setBody("foo"));
+
+    TestInterface api = Feign.builder()
+        .logger(new Logger(){
```

Comment:
j.u.Logger, SLF4J and others conditionally call logAndRebufferResponse based on the framework-specific log level. That's why there's an anonymous logger specified here.